shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -706,6 +706,14 @@ static bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
       int64_t current_platform_view_id = composition_order_[j - 1];
       SkRect platform_view_rect = GetPlatformViewRect(current_platform_view_id);
       std::vector<SkIRect> intersection_rects = slice->region(platform_view_rect).getRects();
+      for (auto it = intersection_rects.begin(); it != intersection_rects.end(); /*no-op*/) {
+        if (it->width() <= 1 || it->height() <= 1) {
```

Comment:
So if the intersection is less than a pixel in either direction, we ignore it? I suppose this makes sense - its why we were seeing those 1 pixel boundaries.

Reply:
If the intersection values you are seeing are even smaller than 1px, you could tighten the condition further (i.e. 0.5 or 0.25 or 0.1) but that might be unecessary.

Reply:
We are getting 1 pixel because of a [roundOut](https://api.skia.org/structSkRect.html#a8a82140d34549c4add33cd988c7dded8) operation. Note that this is already in physical pixels in integers and won't have 0.5 here.

Reply:
Ahh, where does the round out happen?

Reply:
The `roundOut` happens to both flutter layer and platform view layer. For flutter layer, it's https://github.com/flutter/engine/blob/main/display_list/geometry/dl_rtree.cc#L209, for platform view layer, it's https://github.com/flutter/engine/blob/main/flow/embedded_views.h#L343

For example, if we interleave flutter widget and platform view in a list, and let's say we have a flutter widget (top = 0, bottom = 100.1), and a platform view below that widget (top = 100.1, bottom = 200). They are NOT supposed to be overlapping. However, after rounding out, we will get flutter widget (top = 0, bottom = 101), and platform view (top = 100, bottom 200). This will result in 1 pixel overlap as long as there's a floating point in the coord.

Reply:
This might have a much wider impact then. I wonder if it is strictly necessary that we round out there instead of using fp bounds? FYI @flar .

Reply:
I'm not saying this would be better with pixel aware layouts. I'm saying that there is no solution at this level to the problem. The only solution that doesn't cause random side effects for other test cases is pixel-aware layouts.

We do pixel aware layouts, or we live with this problem.

Reply:
Coming up with a threshold might improve the performance of this one test case, but an app that has .15 overlap would fail to benefit.

This is a solution for a particular example that is neither generalized to all cases like this nor is it without side effects.

Reply:
This particular problem is scrolling though, which is a very general problem. We are treating platform views and flutter Ui as _always_ overlapping, even if that overlap is a near infinitesimal

Reply:
How is scrolling any different? It's just a layout. Scroll to pixel boundaries and the problem disappears...

Reply:
Adding a 1 unit pad between the widget and the platform view would also prevent them from overlapping. Regardless of DPI, 1 unit in the framework layout coordinates will always be at least 1 pixel. That doesn't require any engine changes, just add a Padding around the widget or the PV.

Reply:
This issue is a scrolling issue. Have you looked at the benchmark application?

Reply:
To clarify, the overlap between (0-100.1) and (100.1-200) is 0, not 0.1. The reason we get 1 pixel overlay is because we computed the intersection AFTER we `roundOut` both sizes. 

An alternative is to compute the intersection in the original floating point values (without roundOut).